rfc/rfc006-distributed-registry.md

```diff
@@ -379,7 +380,79 @@ The reference MUST resolve to a contact information service as described above.
 Since the information is self-proclaimed and not authenticated or verified in any way, applications MUST treat it as
 untrusted, with great care. Failing to do so could make the operator of the node vulnerable for spoofing and other attacks. 
 
-## 5. Supported Cryptographic Algorithms and Key Types
+## 5. Conflict resolution
+
+§3.4 of [RFC004](rfc004-verifiable-transactional-graph.md) requires each payload type to describe how conflicts are resolved when parallel transactions are encountered.
+A DID Document is not a CRDT, but its contents are. The paragraphs below describe how each of the elements should be treated in case of a conflict.
+First we describe the mechanism of detecting and resolving conflicts. DID Documents do not refer to their previous version in any way. 
+The transactions described in [RFC004](rfc004-verifiable-transactional-graph.md) do refer to previous transactions. This system can be used to detect conflicts and to resolve them.
+For DID Documents additional transactional requirements are added:
+
+* Updates to DID Documents MUST refer in their transaction to the transaction of the previous version. They SHOULD also refer to the latest transactions as described by [RFC005](rfc005-distributed-network-using-grpc.md).
+* In case of a conflict, the conflict can be resolved with an additional transaction referring to the all the conflicting transactions.
+
+## 5.1 Conflict detection
+
+The algorithm for conflict detection can be summarized as follows:
+
+- A DID Document update transaction is received.
+- The current version is resolved together with the metadata (the metadata holds the originating transaction(s)).
+- If the transaction refers to all transactions present in the metadata, the update is valid, and the DID Document is replaced.
+- If the transaction does not refer to all transactions, it is a conflict, and the rules from the paragraphs below are applied. 
+
+A transaction is modelled as:
+
+```
+hash // sha256(data)
+data // A signed JWS in compact form, here it'll represent a DID Document. It also contains the `prevs` field which refers to previous transactions.
+```
+
+The `hash` values of the transactions are stored within the metadata of a DID Document.
```

Comment:
Needs clarification; what is "the metadata of the DID Document"? We have metadata that's derived when a DID Document is resolved, but it's not stored in a spec'd data format (as described by RFC004). Whether it's stored or not is up to the implementor? Or do you refer to the pseudocode below?

Reply:
changed to generic requirement of storing hash values next to DID Document